### ISSUE TITLE
Builds systematically fail with custom docker image

### ISSUE THREAD

**Issue by gtnx** [Shippable/support#4787]
### Description of your issue:

Builds using a custom docker image (eg piu-pollux) always fail. Last image upload was 1 month ago but it fails since yesterday.

```bash
  wait_for_exit
  echo Container c.exec.piu-pollux.19851.1 exited with 100
Container c.exec.piu-pollux.19851.1 exited with 100
  sudo docker logs c.exec.piu-pollux.19851.1
/home/shippable/cexec/build.sh: line 20: lsb_release: command not found
********************************************************************
*                           WARNING                                *
*                                                                  *
* You are not running Ubuntu. This configuration is not supported. *
*                                                                  *
********************************************************************
Looking for apt-get...
Found apt-get at /usr/bin/apt-get
Looking for sudo...
Found sudo at /usr/bin/sudo
Looking for git...
Found git at /usr/bin/git
Looking for ssh-agent
Could not find ssh-agent. Installing...
Ign http://ftp.fr.debian.org jessie-backports InRelease
Get:1 https://artifacts.elastic.co stable InRelease [78 B]
Ign https://artifacts.elastic.co stable InRelease
Get:2 http://apt.postgresql.org jessie-pgdg InRelease [56.5 kB]
Ign http://deb.debian.org jessie InRelease
Ign http://deb.debian.org jessie-updates InRelease
Get:3 https://artifacts.elastic.co stable Release.gpg [473 B]
Get:4 https://artifacts.elastic.co stable Release [5116 B]
Get:5 http://deb.debian.org jessie Release.gpg [2420 B]
Get:6 http://security.debian.org jessie/updates InRelease [44.9 kB]
Ign http://deb.debian.org jessie-updates Release.gpg
Get:7 http://deb.debian.org jessie Release [148 kB]
Ign http://deb.debian.org jessie-updates Release
Err http://deb.debian.org jessie-updates/main amd64 Packages
  
Ign http://ftp.fr.debian.org jessie-backports Release.gpg
Err http://deb.debian.org jessie-updates/main amd64 Packages
  
Err http://deb.debian.org jessie-updates/main amd64 Packages
  
Err http://deb.debian.org jessie-updates/main amd64 Packages
  
Err http://deb.debian.org jessie-updates/main amd64 Packages
  404  Not Found
Get:8 https://artifacts.elastic.co stable/main amd64 Packages [26.6 kB]
Ign http://ftp.fr.debian.org jessie-backports Release
Get:9 http://apt.postgresql.org jessie-pgdg/main amd64 Packages [200 kB]
Get:10 http://deb.debian.org jessie/main amd64 Packages [9098 kB]
Get:11 http://security.debian.org jessie/updates/main amd64 Packages [822 kB]
Err http://ftp.fr.debian.org jessie-backports/main amd64 Packages
  
Err http://ftp.fr.debian.org jessie-backports/main amd64 Packages
  
Err http://ftp.fr.debian.org jessie-backports/main amd64 Packages
  
Err http://ftp.fr.debian.org jessie-backports/main amd64 Packages
  
Err http://ftp.fr.debian.org jessie-backports/main amd64 Packages
  404  Not Found
W: Failed to fetch http://deb.debian.org/debian/dists/jessie-updates/main/binary-amd64/Packages  404  Not Found
Fetched 10.4 MB in 3s (3228 kB/s)
W: Failed to fetch http://ftp.fr.debian.org/debian/dists/jessie-backports/main/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

- https://app.shippable.com/github/optimdata/piu-pollux/runs/19851/1/console

**Comment by a-murphy** [comment]
Since your image doesn't have `ssh-agent`, we're trying to install it.  The `apt-get update` before installing the `openssh-client` package is failing.  Since this is your own image, I would suggest updating the image by either updating or removing the sources that are no longer available.  You could also try to install `ssh-agent` yourself so that we won't run `apt-get update` in that location,

**Comment by clippermadness** [comment]
@gtnx I think the issue is that your custom image is based on Debian 8 Jessie. The packages for that platform have just recently been moved to the Debian archives:
https://lists.debian.org/debian-devel-announce/2019/03/msg00006.html

Hence the 404 when downloading package updates.

We saw the same issue because our custom image was based on the `node:8.4` image. When we changed that to `node:8.4-stretch`, everything worked fine.
